batchflow/models/torch/base.py

```diff
@@ -338,18 +371,29 @@ def _make_optimizer(self, config):
         if decay_name is not None:
             self.lr_decay = decay_name(self.optimizer, **decay_args)
 
+    def _update_decay_args(self, decay_name, decay_args):
+        decay_dict = DECAYS_DEFAULTS.get(decay_name).copy()
+        decay_dict.update(decay_args)
+        return decay_dict
+
     def _make_decay(self, config):
         decay_name, decay_args = unpack_fn_from_config('decay', config)
 
+        if 'num_iters' not in config:
+            raise ValueError('Missing required key ```num_iters``` in the cofiguration dict.')
+        self.num_iters = config.pop('num_iters')
+        DECAYS_DEFAULTS[DECAYS['cos']].update(T_max=self.num_iters)
+
         if decay_name is None or callable(decay_name) or isinstance(decay_name, type):
             pass
         elif isinstance(decay_name, str) and hasattr(torch.optim.lr_scheduler, decay_name):
             decay_name = getattr(torch.optim.lr_scheduler, decay_name)
         elif decay_name in DECAYS:
-            decay_name = DECAYS.get(re.sub('[-_ ]', '', decay_name).lower(), None)
+            decay_name = DECAYS.get(decay_name)
         else:
             raise ValueError("Unknown learning rate decay method", decay_name)
-
+        if decay_name in DECAYS_DEFAULTS:
+            decay_args = self._update_decay_args(decay_name, decay_args)
```

Comment:
What's the point in moving 2 lines of code to a separate method?

Reply:
You don't seem to use anywhere else

Reply:
In line #385 you update DEFAULT_DICT, while here you update the copy of DEFAULT_DICT, why?